bitnami/postgresql-repmgr/12/debian-12/rootfs/opt/bitnami/scripts/librepmgr.sh

```diff
@@ -864,6 +860,9 @@ repmgr_initialize() {
     fi
     if ! repmgr_is_file_external "pg_hba.conf"; then
         is_boolean_yes "$REPMGR_PGHBA_TRUST_ALL" || postgresql_restrict_pghba
+    else
+        cp "${REPMGR_MOUNTED_CONF_DIR}/pg_hba.conf" "${POSTGRESQL_MOUNTED_CONF_DIR}/pg_hba.conf"
+        cp "${REPMGR_MOUNTED_CONF_DIR}/pg_hba.conf" "${POSTGRESQL_CONF_DIR}/pg_hba.conf"
```

Comment:
In this case, you should not copy the file to `POSTGRESQL_MOUNTED_CONF_DIR` and copy it directly to `"${POSTGRESQL_CONF_DIR}/pg_hba.conf"`.

Files are copied from `POSTGRESQL_MOUNTED_CONF_DIR` to `POSTGRESQL_CONF_DIR` during the execution of `postgresql_initialize`.

If the container/pod were restarted (not recreated) it may persist the file at `POSTGRESQL_MOUNTED_CONF_DIR` causing the original issue you reported again.

```suggestion
        cp "${REPMGR_MOUNTED_CONF_DIR}/pg_hba.conf" "${POSTGRESQL_CONF_DIR}/pg_hba.conf"
```
This suggestion should also be applied to all the copies of this file.

Reply:
Thanks for your suggestion, I've removed it on 35cd8b0753d03fdc1db98f0962a07f2dfdb9d29f